Node-DC-EIS-client/runspec.py

```diff
@@ -754,29 +821,242 @@ def send_request(employee_idlist):
+          url_type = 1
+          tot_get = tot_get +1
+          if not(parsed.path == "/employees/zipcode" or parsed.path == "/employees/name"):
+              ids = getNextEmployeeId()
+              url = url+ids
+        if(urllist[execute_request.url_index]['method']== 'POST'):
+          url_type = 2
+          tot_post = tot_post +1
+        if(urllist[execute_request.url_index]['method']== 'DELETE'):
+          url_type = 3
+          tot_del = tot_del +1
+          ids = getNextEmployeeId()
+          removeEmployeeId(ids)
+          url = url+ids
+        if(int(concurrency) == 1):
+          main_entry(url,execute_request.request_index,url_type,log_dir,phase,interval,run_mode,temp_log)
+        else:
+          pool.spawn_n(main_entry,url,execute_request.request_index,url_type,log_dir,phase,interval,run_mode,temp_log)
+        execute_request.request_index += 1
+        execute_request.url_index += 1
+         
+    except Exception, err:
+      traceback.print_exc()
+      print Exception, err
+      exit(1)
+
+# Initializing the Static Variables of the Function , which are used as counter
+execute_request.request_index = 1 
+execute_request.url_index = 0
+
+def timebased_run(pool):
+  """
+  # Desc  : Function to start time based run
+  #         Uses threadpool for concurrency, and sends concurrent requests
+  #         to server for the input time interval.
+  #         Dynamically generates URL with employeeid, zipcode, name for GET
+  #         requests
+  # Input : threadpool with concurrency number of threads
+  # Output: Generates per request details in a templog file
+  """
+  global after_run
+  global phase
+  global MT_interval
+  global rampup_rampdown
+  global requests_done
+  global tot_get
+  global tot_post
+  global tot_del
+  global log
+  global log_dir
+  global interval
+  global temp_log
+  global output_file
+  global processing_complete
+  url_index = 0
+  request_index = 0 # Initializing the Request Counter to 0
+
+  #Spin Another Process to do processing of Data
+  post_processing = Process(target=process_time_based_output,args=(log_dir,interval,rampup_rampdown,MT_interval,temp_log,output_file,memlogfile,instance_id,multiple_instance))
+  post_processing.start()
+  print ("[%s] Starting time based run." % (getCurrentTime()))
+  if ramp:
+    phase = "RU"
+    start = time.time()
+    print("[%s] Entering RampUp time window." % (getCurrentTime()))
+  else:
+    phase = "MT"
+    start = time.time()
+    print("Entering Measuring time window : [%s]" % (time.strftime("%d-%m-%Y %H:%M:%S")))
+    util.record_start_time()
+  print ("[%s] Started processing of requests with concurrency of [%d] for [%d] seconds" % (time.strftime("%d-%m-%Y %H:%M:%S"), int(concurrency), int(MT_interval)))
+  #print ("[%s] Started processing of [%d] total requests with concurrency of [%d]" % (time.strftime("%d-%m-%Y %H:%M:%S"), int(request), int(concurrency)))
+  if sys.platform != "win32":
```

Comment:
Looks like the else part is for windows. I'm just not sure why we need signals. It seems like the windows version would work for both.